app/controllers/decidim/omniauth_registrations_controller_override.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Decidim
+  module OmniauthRegistrationsControllerOverride
+    extend ActiveSupport::Concern
+
+    included do
+      include Decidim::AfterSignInActionHelper
+
+      def after_sign_in_path_for(user)
+        after_sign_in_action_for(user, request.params[:after_action]) if request.params[:after_action].present?
```

Comment:
I wonder if we should add the return otherwise it continues until the end of function
```suggestion
        return after_sign_in_action_for(user, request.params[:after_action]) if request.params[:after_action].present?
```

Reply:
The `after_sign_in_action_for` doesn't manage the redirection, only the action to be process before the redirection happen.  
We need the function to continue its process.